Doc/library/typing.rst

```diff
@@ -1266,213 +1323,325 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
-       class StringSubclass(str):
-           pass
+      x = print_capitalized('a string')
+      reveal_type(x)  # revealed type is str
 
-       y = print_capitalized(StringSubclass('another string'))
-       reveal_type(y)  # revealed type is StringSubclass
+      class StringSubclass(str):
+          pass
 
-       z = print_capitalized(45)  # error: int is not a subtype of str
+      y = print_capitalized(StringSubclass('another string'))
+      reveal_type(y)  # revealed type is StringSubclass
 
-    Type variables can be bound to concrete types, abstract types (ABCs or
-    protocols), and even unions of types::
+      z = print_capitalized(45)  # error: int is not a subtype of str
 
-       U = TypeVar('U', bound=str|bytes)  # Can be any subtype of the union str|bytes
-       V = TypeVar('V', bound=SupportsAbs)  # Can be anything with an __abs__ method
+   Type variables can be bound to concrete types, abstract types (ABCs or
+   protocols), and even unions of types::
 
-.. _typing-constrained-typevar:
+      U = TypeVar('U', bound=str|bytes)  # Can be any subtype of the union str|bytes
+      V = TypeVar('V', bound=SupportsAbs)  # Can be anything with an __abs__ method
 
-    Using a *constrained* type variable, however, means that the ``TypeVar``
-    can only ever be solved as being exactly one of the constraints given::
+      # Can be anything with an __abs__ method
+      def print_abs[V: SupportsAbs](arg: V) -> None:
+          print("Absolute value:", abs(arg))
 
-       a = concatenate('one', 'two')
-       reveal_type(a)  # revealed type is str
+   .. _typing-constrained-typevar:
 
-       b = concatenate(StringSubclass('one'), StringSubclass('two'))
-       reveal_type(b)  # revealed type is str, despite StringSubclass being passed in
+   Using a *constrained* type variable, however, means that the ``TypeVar``
+   can only ever be solved as being exactly one of the constraints given::
 
-       c = concatenate('one', b'two')  # error: type variable 'A' can be either str or bytes in a function call, but not both
+      a = concatenate('one', 'two')
+      reveal_type(a)  # revealed type is str
 
-    At runtime, ``isinstance(x, T)`` will raise :exc:`TypeError`.  In general,
-    :func:`isinstance` and :func:`issubclass` should not be used with types.
+      b = concatenate(StringSubclass('one'), StringSubclass('two'))
+      reveal_type(b)  # revealed type is str, despite StringSubclass being passed in
 
-    Type variables may be marked covariant or contravariant by passing
-    ``covariant=True`` or ``contravariant=True``.  See :pep:`484` for more
-    details.  By default, type variables are invariant.
+      c = concatenate('one', b'two')  # error: type variable 'A' can be either str or bytes in a function call, but not both
 
-.. class:: TypeVarTuple
+   At runtime, ``isinstance(x, T)`` will raise :exc:`TypeError`.  In general,
+   :func:`isinstance` and :func:`issubclass` should not be used with types.
```

Comment:
I know we already have this wording in `typing.rst`, but I don't understand the meaning of:

> In general, `isinstance()` and `issubclass()` should not be used with types.

Reply:
I think "types" is supposed to mean "static types", i.e. things you put in type annotations. The distinction between "classes" and "types" was controversial when PEP 484 was discussed. Since then, we've blurred that distinction with e.g. PEP 604. We can probably remove this sentence here, as it's not directly relevant to TypeVar.

Reply:
Yeah, let's axe it.